### PR TITLE
feat(rds/instance): rds modify status check retry logic

### PR DESCRIPTION
**Pull request by wanglin100**
**What this PR does / why we need it**:

rds modify status check retry logic

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: huaweicloud/services/rds/resource_huaweicloud_rds_instance.go [hunk 1]
```diff
@@ -414,8 +413,21 @@ func resourceRdsInstanceCreate(ctx context.Context, d *schema.ResourceData, meta
 		// Ensure that the instance is 'ACTIVE', not going to enter 'BACKING UP'.
 		ContinuousTargetOccurence: 2,
 	}
-	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
-		return diag.Errorf("error waiting for RDS instance (%s) creation completed: %s", instanceID, err)
+	retryErr := resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		_, status, err := stateConf.Refresh()
+		if err == nil {
+			if status == "FAILED" {
+				return resource.NonRetryableError(fmt.Errorf("instance status: %s", status))
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_instance.go]: can we add this judgement to the refresh function?

Comment (wanglin100) [reply huaweicloud/services/rds/resource_huaweicloud_rds_instance.go]: I have modified.

File: huaweicloud/services/rds/resource_huaweicloud_rds_instance.go [hunk 2]
```diff
@@ -1203,7 +1202,9 @@ func rdsInstanceStateRefreshFunc(client *golangsdk.ServiceClient, instanceID str
 		if instance.Id == "" {
 			return instance, "DELETED", nil
 		}
-
+		if instance.Status == "FAILED" {
+			return nil, instance.Status, fmt.Errorf("RDS instance creation failed. ")
```
Comment (ShiChangkuo) [inline huaweicloud/services/rds/resource_huaweicloud_rds_instance.go]:
```suggestion
			return nil, instance.Status, fmt.Errorf("RDS instance creation failed")
```

Comment (wanglin100) [reply huaweicloud/services/rds/resource_huaweicloud_rds_instance.go]: ok, i have modified

**Pull request merged**